### PR TITLE
Simplify extensability of BaseHiveConnectorTest

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

**Comment by losipiuk** [comment]
cc: @arhimondr

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFaultTolerantExecutionConnectorTest.java [hunk 1]
```diff
@@ -14,15 +14,18 @@
...
 public class TestHiveFaultTolerantExecutionConnectorTest
         extends BaseHiveConnectorTest
 {
-    public TestHiveFaultTolerantExecutionConnectorTest()
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
     {
-        super(FaultTolerantExecutionConnectorTestHelper.getExtraProperties());
+        return BaseHiveConnectorTest.createHiveQueryRunner(FaultTolerantExecutionConnectorTestHelper.getExtraProperties());
```
Comment (Praveen2112) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFaultTolerantExecutionConnectorTest.java:28]: nit: static import

**Review by Praveen2112 — APPROVED** [review]

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFaultTolerantExecutionConnectorTest.java (hunk 1 above)
Comment (losipiuk) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFaultTolerantExecutionConnectorTest.java:28]: I would rather leave it as is to bot make it confusable with non-static `getQueryRunner` method

Comment (findepi) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFaultTolerantExecutionConnectorTest.java:28]: Maybe make it non-static and `super.` here?

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHivePipelinedExecutionConnectorTest.java [hunk 2]
```diff
@@ -14,12 +14,15 @@
 package io.trino.plugin.hive;
 
 import com.google.common.collect.ImmutableMap;
+import io.trino.testing.QueryRunner;
 
 public class TestHivePipelinedExecutionConnectorTest
```
Comment (findepi) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHivePipelinedExecutionConnectorTest.java:19]: what does "pipelined execution" mean here?

it this is "test hive with defaults", let's call the class just `TestHiveConnectorTest`

**Review by findepi — APPROVED** [review]

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHivePipelinedExecutionConnectorTest.java (hunk 2 above)
Comment (losipiuk) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHivePipelinedExecutionConnectorTest.java:19]: Yeah - effectively it is with defaults (no failure recovery). I will rename in a separate PR after this gets merged.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFaultTolerantExecutionConnectorTest.java (hunk 1 above)
Comment (losipiuk) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestHiveFaultTolerantExecutionConnectorTest.java:28]: IDK. It felt nicer to me to keep the contract that to extend the abstract the `BaseHiveConnectorTest` it is a sole responsibility of leaf class to provide implementation of `createQueryRunner`. 
In that case keeping `BaseHiveConnectorTest.createHiveQueryRunner` as static makes more sense. Then `BaseHiveConnectorTest` serves just as a namespace for helper `createHiveQueryRunner` to be used by the interested parties. The fact it is part of the hierarchy does not really matter.

If you do not feel strongly I would keep it.

**Pull request merged**